beacon-chain/rpc/service.go

```diff
@@ -598,6 +611,49 @@ func (s *Service) Start() {
 	}()
 }
 
+func updateTrackValidatorCacheWithProposerSettings(ctx context.Context, chain blockchain.ChainInfoFetcher, settings *validatorpb.ProposerSettingsPayload, tackedValidatorCache *cache.TrackedValidatorsCache) error {
```

Comment:
Feels like there's some kind of code smell here for getting the index of the validator....